marimo/_server/file_router.py

```diff
@@ -175,25 +190,34 @@ def _load_files(self) -> List[MarimoFile]:
                 continue
             # Iterate over all files in the directory
             for filename in filenames:
-                if not filename.endswith(".py"):
+                if not any(
+                    filename.endswith(ext) for ext in allowed_extensions
+                ):
                     continue
                 full_path = os.path.join(root, filename)
                 relative_path = os.path.relpath(full_path, directory)
-                try:
-                    with open(full_path, "r", encoding="utf-8") as f:
-                        if "marimo.App" in f.read():
-                            files.append(
-                                MarimoFile(
-                                    name=filename,
-                                    path=relative_path,
-                                    last_modified=os.path.getmtime(full_path),
-                                )
-                            )
-                except Exception as e:
-                    LOGGER.debug("Error reading file %s: %s", full_path, e)
+                # Python files must contain "marimo.App", or markdown files
```

Comment:
Could cue on `marimo-version: ` for #1332

Reply:
ill do this in a followup which may touch this code, but good call